ci/nightly-build.sh

```diff
@@ -26,7 +26,8 @@ MVN="mvn -Dmaven.wagon.http.retryHandler.count=3 -B"
 CUDA_VER=${CUDA_VER:-cuda`nvcc --version | sed -n 's/^.*release \([0-9]\+\)\..*$/\1/p'`}
 PARALLEL_LEVEL=${PARALLEL_LEVEL:-4}
 USE_GDS=${USE_GDS:-ON}
-${MVN} clean package ${MVN_MIRROR}  \
+SANITIZER_CMD="compute-sanitizer --tool memcheck --target-processes all $SANITIZER_OPTS"
```

Comment:
In the past we've needed to specify a hefty `--launch-timeout` to prevent compute-sanitizer from giving up on instrumenting a process that doesn't use the GPU shortly after startup.  That would certainly be the case here, as it could take close to an hour to complete building.  cc: @abellina who has more experience with this.

It would be nice to not have compute-sanitizer running for the main compiliation part of the build -- it's just a waste.  A rebuild without cleaning is pretty fast, and we could skip the Java tests on the first "clean package" build, but I don't know if there's a way to skip running the C++ tests on the first build and only on the subsequent test build.

Reply:
+1 on not having this on during any part of compilation. It should only be used when running the tests. I also wouldn't run all the tests as it is likely to take a very long time.

Yes the `--launch-timeout` can be really high and it was needed to wait for that first cuda call, it is likely a requirement in every scenario (the default is 60 seconds for this case).

I do not know of a good way to have maven surefire replace JAVA_HOME or wrap its test process with this (there is also attach mode, but that seems complicated to script). I also think that we don't want to run this for all tests because it will be very slow, and it will alter timings that could hide other synchronization issues. 

I think we may need to curate the tests and run this only for certain things with a pytest mark. I would skip scala tests and do it all in the integration tests, also the integration tests should make this a lot easier to be opt in and configurable. Thoughts?

Reply:
> I think we may need to curate the tests and run this only for certain things with a pytest mark. I would skip scala tests and do it all in the integration tests, also the integration tests should make this a lot easier to be opt in and configurable. Thoughts?

This is spark-rapids-jni, there's no Python tests.  I think there's a small enough number of tests (and they're quick enough) to warrant running them on all the tests.  However it would be good to know how much slower this makes the test run.

Reply:
Is it possible to use the [toolchain](https://maven.apache.org/surefire/maven-surefire-plugin/java9.html#head3) to replace the JAVA_HOME